apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -2742,15 +2826,16 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
     :ok
   end
 
-  defp send_consumer_properties_payload(realm, device_id, abs_paths_list) do
+  defp send_consumer_properties_payload(realm, device_id, abs_paths_list, compression_format) do
     topic = "#{realm}/#{Device.encode_device_id(device_id)}/control/consumer/properties"
 
     uncompressed_payload = Enum.join(abs_paths_list, ";")
 
-    payload_size = byte_size(uncompressed_payload)
-    compressed_payload = :zlib.compress(uncompressed_payload)
-
-    payload = <<payload_size::unsigned-big-integer-size(32), compressed_payload::binary>>
+    payload =
+      case compression_format do
+        :zlib -> :zlib.compress(uncompressed_payload)
```

Comment:
In this case, the header should be kept as it is part of the already existing protocol: the device expects to receive the uncompressed payload length as header (see https://docs.astarte-platform.org/astarte/latest/080-mqtt-v1-protocol.html#purge-properties)
```suggestion
        :zlib -> 
          payload_size = byte_size(uncompressed_payload)
          compressed_payload = :zlib.compress(uncompressed_payload)
          <<payload_size::unsigned-big-integer-size(32), compressed_payload::binary>>
```